tests/rptest/tests/pandaproxy_test.py

```diff
@@ -223,10 +225,10 @@ def _get_kafka_cli_tools(self):
                              user=cfg.get('sasl_plain_username'),
                              passwd=cfg.get('sasl_plain_password'))
 
-    def _base_uri(self, hostname=None):
+    def _base_uri(self, hostname=None, tls_enabled: bool = False):
         hostname = hostname if hostname else self.redpanda.nodes[
             0].account.hostname
-        return f"http://{hostname}:8082"
+        return f"https://{hostname}:8082" if tls_enabled else f"http://{hostname}:8082"
```

Comment:
nit: This might be slightly easier to read:
```suggestion
        scheme = "https" if tls_enabled else "http"
        return f"{scheme}://{hostname}:8082"
```